nav2_map_server/src/map_saver.cpp

```diff
@@ -54,8 +54,8 @@ MapSaver::MapSaver(const rclcpp::NodeOptions & options)
   Magick::InitializeMagick(nullptr);
   {
     mapname_ = declare_parameter("output_file_no_ext", "map");
-    if (mapname_.empty()) {
-      throw std::runtime_error("Map name not provided");
+    if (mapname_ == "map") {
+      mapname_ += "_" + std::to_string(static_cast<int>(now().seconds()));
```

Comment:
This is using the ros clock, which means it could end up with the sim_time timestamp added. I'm not sure if it is a problem in practice, but it seems like this should use the wall clock to ensure a truly unique timestamp.

I'd also consider formatting the time into human readable string like 2019-10-21T13:33:39. That way someone can look at the filenames and kind of have an idea of which file might be which. The code to create a formatted string looks relatively straightforward. There's an example here:
https://stackoverflow.com/questions/9527960/how-do-i-construct-an-iso-8601-datetime-in-c

Reply:
@crdelsey so the sim timestamp isn't really a problem, its not really about what the value is as much as the relative times and that we're not overriding values.

In terms of string, I prefer unix time that way there's no confusion about timezones which can often vary when you have many, many robots deployed across the world